source/common/http/http1/conn_pool.cc

```diff
@@ -155,15 +155,17 @@ void ConnPoolImpl::onConnectionEvent(ActiveClient& client, Network::ConnectionEv
     if (check_for_drained) {
       checkForDrained();
     }
-  } else if (event == Network::ConnectionEvent::Connected) {
-    conn_connect_ms_->complete();
-    processIdleClient(client);
   }
 
   if (client.connect_timer_) {
     client.connect_timer_->disableTimer();
     client.connect_timer_.reset();
   }
+
+  if (event == Network::ConnectionEvent::Connected) {
```

Comment:
Can you elaborate a little on this fix? Seems to me that that the main thing that has happened is this bock was reordered with the connect_timer reset. How does this change things?

Reply:
Will add a code comment.